Assets/Scripts/UI/Text/TextController.cs

```diff
@@ -5,28 +5,46 @@
 
 public class TextController : MonoBehaviour
 {
-    public int velocityMultiplier = 10;
+    public int velocityMultiplier = 3;
 
     public Text timerText;
     public Text positionText;
     public Text speedText;
 
-    private int playerIndex;
+    private Rigidbody playerRB;
+    private PositionTracker positionTracker;
 
     public enum TextTypes{ timer, position, speed};
 
+    private int numRacers;
+
     // Start is called before the first frame update
     void Start()
     {
         ResetTexts();
-        playerIndex = GetPlayerIndex();
+        ResetPlayerReferences();
+        numRacers = RacerManager.GetRacers().Count;
     }
 
     public void Update()
     {
-        SetText(TextTypes.position, RacerManager.GetPositionTracker(playerIndex).GetPosition().ToString());
-        SetText(TextTypes.speed, ((int)(RacerManager.GetRigidbody(playerIndex).velocity.magnitude * velocityMultiplier)).ToString());
-        SetText(TextTypes.timer, RacerManager.GetTimeElapsed().ToString("F2"));
+        if (numRacers != RacerManager.GetRacers().Count) 
+        {
+            ResetPlayerReferences();
```

Comment:
You don't need to reset player references every time a racer is removed from the race